tests/unit/test_dataset_manager.py

```diff
@@ -153,3 +171,31 @@ def test_transform_skip_prepare_input_files(manager_class):
         dm.populate_metadata.assert_called_once_with()
         dm.prepare_input_files.assert_not_called()
         dm.create_zarr_json.assert_called_once_with()
+
+    @staticmethod
+    def test_get_subclasses(manager_class):
+        assert set(manager_class.get_subclasses()) == {John, Paul, George, Ringo}
+        assert set(John.get_subclasses()) == {George, Ringo}
+        assert set(Paul.get_subclasses()) == {George, Ringo}
+        assert set(George.get_subclasses()) == {Ringo}
+        assert set(Ringo.get_subclasses()) == set()
+
+    @staticmethod
+    def test_get_subclass(manager_class):
+        assert manager_class.get_subclass("John") is John
+        assert manager_class.get_subclass("Paul") is Paul
+        assert manager_class.get_subclass("George") is George
+        assert manager_class.get_subclass("Ringo") is Ringo
+
+        assert John.get_subclass("George") is George
+        assert John.get_subclass("Ringo") is Ringo
+
+        assert Paul.get_subclass("George") is George
+        assert Paul.get_subclass("Ringo") is Ringo
+
+        assert George.get_subclass("Ringo") is Ringo
+
+    @staticmethod
+    def test_get_subclass_not_found():
```

Comment:
perhaps for good measure assert that `manager_class.get_subclass(“Pete”)` raises an error

Reply:
Well, returns None and raises a warning, but sure. 32a76ba